<?xml version="1.0"?>
<launch>

<arg name="stamped" default="true" description="publish stamped or not-stamped twist" />

<!-- launch-prefix="xterm -e" -->
<node pkg="uos_diffdrive_teleop" 
    exec="uos_diffdrive_teleop_key" 
    name="uos_diffdrive_teleop_key" 
    output="screen"
    launch-prefix="xterm -e">
    
    <param name="use_sim_time" value="True" />
    <param name="max_vel" value="1.0"/>
    <param name="max_rot_vel" value="2.0"/>

    <param name="use_button_pressure" value="false"/>

    <!-- m/s² -->
    <param name="acc_ahead_pos" value="2.2"/>
    <param name="acc_ahead_neg" value="2.4"/>
    <param name="acc_ahead_stop" value="4.0"/>

    <!-- rad/s² -->
    <param name="acc_rot_pos" value="2.6"/>
    <param name="acc_rot_neg" value="2.8"/>
    <param name="acc_rot_stop" value="4.6"/>

    <param name="use_stamped_twist" value="$(var stamped)" />
    <param name="cmd_frame" value="base_link" />
</node>
</launch>
